pubsub/README.rst

```diff
@@ -113,9 +111,35 @@ Here's a few hints:
     same as ``ack_window`` but for nack requests
 
 
-``subscribe`` has also an optional ``metrics_client`` argument. You can provide
-any metrics agent that implements the same interface as ``MetricsAgent``
-(Datadog client will do ;) ) and get the following metrics:
+Prometheus Metrics
+^^^^^^^^^^^^^^^^^^
+
+If you like pull-based metrics like Prometheus you will be pleased to know that
+the subscriber records Prometheus metrics in the form
+``gcloud_aio_pubsub_<metric>``, which will have no effect if you don't use
+Prometheus to scrape app metrics:
+
+- ``subscriber_consume_total`` (labels: ``outcome = {'succeeded', 'cancelled',
+  'failed', 'failfast'}``) - [counter] a consume operation has completed with a
+  given outcome
+- ``subscriber_consume_latency_seconds`` (labels: ``phase = {'receive',
+  'queueing', 'runtime'}``) - [histogram] how many seconds taken to receive a
+  message, while waiting for processing, or to complete the callback
+- ``subscriber_batch_status_total`` (labels: ``component = {'acker', 'nacker'},
+  outcome = {'succeeded', 'failed'}``) - [counter] a batch has succeeded or
+  failed to be acked or nacked
+- ``subscriber_messages_processed_total`` (labels: ``component = {'acker',
```

Comment:
In this case, a total sum over `subscriber_messages_processed_total` doesn't really represent anything meaningful, does it? IIUC, it should at most be 2x the total number of messages that were pulled during a given time period. Does this suggest the metric should be broken up? (based on the guidelines of our [own metric guide](https://docs.talkiq-echelon.talkiq.com/static/docs/guides/observability/metrics.html#metric-grouping)

Reply:
That's a great point, I think that receiving the messages is an entirely different metric. With just the `acker/nacker` IMO this now makes sense since the total messages processed (either via `ack` or `nack`) through the PubSub client can be summed by using this metric. Thoughts?